packages/vscode/src/plugins/nextjs/addJSDoc.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  ArrowFunction,
+  FunctionDeclaration,
+  SourceFile,
+  TypeGuards,
+} from 'ts-morph'
+import { NextFunctionName, NextFunctionType } from './constants'
+
+function buildJSDocType(base: string, paramName: string) {
+  return `{import('next').${base} } ${paramName}`
+}
+function addType(
+  node: ArrowFunction | FunctionDeclaration,
+  foundNextFunctions: NextFunctionType,
+) {
+  const functionNames = Object.keys(foundNextFunctions) as NextFunctionName[]
+  if (functionNames && functionNames.length !== 0) {
+    const params = node.getParameters()
+    const param = params[0]
+    const typeString = foundNextFunctions[functionNames[0]].type
+    const jsDocType = buildJSDocType(typeString, 'props')
+    if (
+      TypeGuards.isFunctionDeclaration(node) &&
+      node.getJsDocs().length === 0
+    ) {
+      if (!param) {
+        node.addParameter({
+          name: 'props',
+        })
+      }
+      node.addJsDoc({
+        tags: [
+          {
+            tagName: 'param',
+            text: jsDocType,
+          },
+        ],
+      })
+    }
+  }
+}
+
+export function addJSDoc(
+  sourceFile: SourceFile,
+  foundNextFunctions: NextFunctionType,
+) {
+  const defaultExportSymbol = sourceFile.getDefaultExportSymbol()
+
+  if (defaultExportSymbol) {
+    // you will need to handle more scenarios than what's shown here
+    const declaration = defaultExportSymbol.getDeclarations()[0]
+    if (!declaration) return
+
+    if (TypeGuards.isFunctionDeclaration(declaration)) {
+      addType(declaration, foundNextFunctions)
+    } else if (TypeGuards.isExportAssignment(declaration)) {
+      const expr = declaration.getExpression()
+
+      if (TypeGuards.isArrowFunction(expr)) {
+        addType(expr, foundNextFunctions)
+      } else if (TypeGuards.isIdentifier(expr)) {
+        const node = expr
+          .findReferences()[0]
+          .getDefinition()
+          .getDeclarationNode()
+        const child = node?.getLastChild()
+
+        if (
+          TypeGuards.isFunctionDeclaration(node) ||
+          TypeGuards.isArrowFunction(node)
+        ) {
+          addType(node, foundNextFunctions)
+        } else if (
+          TypeGuards.isArrowFunction(child) ||
+          TypeGuards.isFunctionDeclaration(child)
+        ) {
+          addType(child, foundNextFunctions)
+        } else {
+          console.log('Unhandled Case 1', child?.getKindName())
```

Comment:
Can you provide some more information on what this unhandled case is? Or some comment at least.

Reply:
It was for debugging, I will remove them